components/kserve/kserve_config_handler.go

```diff
@@ -67,24 +69,25 @@ func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client
 	if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData); err != nil {
 		return fmt.Errorf("error retrieving value for key 'deploy' from configmap %s. %w", KserveConfigMapName, err)
 	}
+	var ingressData map[string]interface{}
+	if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["ingress"]), &ingressData); err != nil {
+		return fmt.Errorf("error retrieving value for key 'ingress' from configmap %s. %w", KserveConfigMapName, err)
+	}
 	modeFound := deployData["defaultDeploymentMode"]
-	if modeFound != string(defaultmode) {
+	ingressCreationValueFound := ingressData["disableIngressCreation"]
+	if (modeFound != string(defaultmode)) || ingressCreationValueFound != disableIngressCreation {
 		deployData["defaultDeploymentMode"] = defaultmode
 		deployDataBytes, err := json.MarshalIndent(deployData, "", " ")
 		if err != nil {
 			return fmt.Errorf("could not set values in configmap %s. %w", KserveConfigMapName, err)
 		}
 		inferenceServiceConfigMap.Data["deploy"] = string(deployDataBytes)
-
-		var ingressData map[string]interface{}
-		if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["ingress"]), &ingressData); err != nil {
-			return fmt.Errorf("error retrieving value for key 'ingress' from configmap %s. %w", KserveConfigMapName, err)
-		}
-		if defaultmode == RawDeployment {
-			ingressData["disableIngressCreation"] = true
-		} else {
-			ingressData["disableIngressCreation"] = false
+		clusterDomain, err := cluster.GetDomain(ctx, cli)
+		if err != nil {
+			return fmt.Errorf("error retrieving cluster domain %s. %w", KserveConfigMapName, err)
 		}
+		ingressData["ingressDomain"] = clusterDomain
```

Comment:
I'd think that we should always ensure that the domain is correctly configured. So, probably this should be outside the `if`.